src/Symfony/Component/Validator/Tests/Constraints/NoSuspiciousCharactersValidatorTest.php

```diff
@@ -56,23 +56,31 @@ public static function provideNonSuspiciousStrings(): iterable
     /**
      * @dataProvider provideSuspiciousStrings
      */
-    public function testSuspiciousStrings(string $string, array $options, string $errorCode, string $errorMessage)
+    public function testSuspiciousStrings(string $string, array $options, array $errors)
     {
         $this->validator->validate($string, new NoSuspiciousCharacters($options));
 
-        $this->buildViolation($errorMessage)
-            ->setCode($errorCode)
+        $violations = $this->buildViolation(reset($errors))
+            ->setCode(key($errors))
             ->setParameter('{{ value }}', '"'.$string.'"')
-            ->assertRaised();
+        ;
+
+        while ($message = next($errors)) {
```

Comment:
`foreach ($errors as $code => $message)` would be a lot more readable IMO (you will need a special handling of the first item s next ones to create `$violations`)